components/match2/commons/match_group_display.lua

```diff
@@ -102,6 +102,42 @@ function MatchGroupDisplay.MatchGroupById(args)
 	})
 end
 
+--[[
+Displays a singleMatch specified by a bracket ID and matchID.
+]]
+function MatchGroupDisplay.MatchByMatchId(args)
+	local bracketId = args.id or args[1]
+	args.id = bracketId
+	args[1] = bracketId
+	assert(bracketId, 'Missing bracket ID')
+
+	local matches = MatchGroupUtil.fetchMatches(bracketId)
+	assert(#matches > 0, 'No data found for bracketId=' .. bracketId)
+
+	if (#matches > 1) then
+		assert(args.matchid, 'Missing match ID')
+		local matchIdLength = string.len(args.matchid)
+		args.matchid = string.rep('0', 4 - matchIdLength) .. args.matchid
```

Comment:
It might be good to include the corresponding logic for brackets

so  

````
{{ShowSingleMatch|meWH7eLN9O|matchid=R03-M001}}

{{ShowSingleMatch|meWH7eLN9O|matchid=R3M1}}
```

both work

Maybe expand MatchGroupUtil.matchIdFromKey to support matchlist id <-> key conversion, and call that

Reply:
Uses MatchGroupUtil.matchIdFromKey now.